docs/manual/philosophy.md

```diff
@@ -0,0 +1,149 @@
+specific programming language - in trying to not make assumptions about
+how software looks or gets packaged. Packaging software can be messy and
+RPM accomodates for that.
+
+It still offers help to create a POSIX like operation system. Marcos
+parameterize locations to make the build instructions more generic and
+more easily transferable between distributions. Nevertheless RPM
+packages are targeting a specific distribution (release). They are
+typically not suited to be installed or even built elsewhere without
+tweaking. RPM acknowledges that different distributions have
+different goals, resulting in different design decisions. The
+specifics of the packages often reflect these.
+
+RPM as a upstream project still tries to keep distributions from
+diverging unnecessarily but is aware that these differences are the
+reason distributions exist in the first place.
+
+## Levels of control
+
+Most things in RPM can be configured by macros. This allows different
+actors to set them and overwrite what was set previously.
+
+Most basic are the defaults delivered by RPM upstream. Distributions
+are able to overwrite them by patching or providing new macro
+files. Packages can also ship macro files for other packages to use as
+BuildRequires. Finally most build related macros can also be
+overwritten in the Spec file - giving the packager the last word.
+
+There are a few command line options to allow users to influence how
+packages are installed but their use is discouraged for the most part
+and packages should be installed as they are.
+
+
+## RPM in the Software handling stack
+
+RPM sees itself as a specific layer in the software handling stack. It
+relies on build systems like make and the upstream provided build
+scripts to orchestrate the actual build and passes the acquiring and
+selection of the "right" packages up to updaters like yum, dnf,
+zypper, etc.
+
+The stack typically looks like this:
+
+ * Upstream build scripts using tools like maker, cmake, ant, ...
+ * rpmbuild for running those via a Spec file
+ * Build systems for installing build dependencies and keeping track of build artifacts
+ * Package repositories offering binary packages
+ * Installers/Updaters selecting and downloading packages
+ * rpm checking for consistency and installing/updating the packages
+
+## Design goals
+
+### Rolling out (security) updates quickly.
+
+Getting updates installed quickly is one of the main design
+goals. Many features work towards this.
+
+#### Packaging dependencies separately
+
+Libraries should be packaged separately and binaries should link to
+the version provided by system packages. Static linking is
+discouraged. This limits the number of packages that need updates or
+re-builds in case of a vulnerability.
+
+#### Unattended operation
+
+Package installation and update is unattended and must not require
+user interaction. When possible delay user interaction to the first
+start up.
```

Comment:
Just drop the second sentence, user interaction is simply not an option during operation.

Reply:
Builds being unattended too doesn't seem to be mentioned anywhere, that's an equally important thing.